ibtk/include/ibtk/FEDataManager.h

```diff
@@ -836,6 +837,52 @@ class FEDataManager : public SAMRAI::tbox::Serializable, public SAMRAI::mesh::St
                                 const std::vector<libMesh::Elem*>& active_elems,
                                 const std::string& system_name);
 
+    /*!
+     * \brief Prolong a side centered value or a density from the FE mesh to the Cartesian
+     * grid.
+     */
+    void prolongData_side(int f_data_idx,
+                          libMesh::NumericVector<double>& F,
+                          libMesh::NumericVector<double>& X,
+                          const std::string& system_name,
+                          bool is_density = true,
+                          bool accumulate_on_grid = true,
+                          bool close_F = true,
+                          bool close_X = true);
```

Comment:
Additionally, these functions are mostly identical. Would you be willing to refactor them into one function?

Reply:
yeah, I can put these into one function.

by free function, where exactly would I put this?  we could also do something like this for the assemble functions for the harmonic and diffusion equation discretizations.  @boyceg mentioned something about this.

Reply:
I think it would be reasonable to create a new header `ibtk/prolongation_and_restriction.h` and source file `ibtk/lagrangian/prolongation_and_restriction.cpp` that contain all of these functions for moving between libMesh and SAMRAI.

Reply:
@drwells @boyceg I combined stuff into a single prolongData function, but the code is a bit clumsy.  There are some local variables which need to be defined at runtime as either Cell or Side Centered Data.  Maybe I should be doing dynamic casting?  At this point, I just declare local variables for both, and use only one, depending on if we are prolonging onto Side or Cell centered data.